src/pdm/cli/commands/venv/activate.py

```diff
@@ -55,6 +56,8 @@ def get_activate_command(self, venv: VirtualEnv) -> str:  # pragma: no cover
             command, filename = "source", "activate"
         activate_script = venv.interpreter.with_name(filename)
         if activate_script.exists():
+            if platform.system() == "Windows":
+                return f"{shlex.quote(str(activate_script))}"
```

Comment:
look above several lines, we have a case for powershell. this will completely bypass it.

Reply:
This only works for powershell 7 and not 5.1. Neither does the old solution work for Batch.

I did not want to remove the previous check with powershell, since you can install powershell on linux (not sure why you would want that but you can) and that supports the dot command.

Output from powershell 5.1
```ps1
> Invoke-Expression (pdm venv activate)
C:\Code\data_indhentning\.venv\Scripts\Activate.ps1 : Cannot dot-source this command because it was defined in a differ
ent language mode. To invoke this command without importing its contents, omit the '.' operator.
At line:1 char:1
+ . 'C:\Code\.venv\Scripts\Activate.ps1'
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [Activate.ps1], NotSupportedException
    + FullyQualifiedErrorId : DotSourceNotSupported,Activate.ps1
```